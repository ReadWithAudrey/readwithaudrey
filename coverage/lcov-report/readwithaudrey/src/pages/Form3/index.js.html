<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/pages/Form3/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/pages/Form3</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Link, graphql } from 'gatsby'
import PropTypes from 'prop-types'
&nbsp;
import {
  Label,
  Button,
  StatusBar,
  RadioButton,
  TextArea,
  TextBox,
  Layout,
} from '../../components/'
&nbsp;
const Form3 = ({ data }) =&gt; {
  const { q1, q2, q3 } = data.markdownRemark.frontmatter
  return (
    &lt;Layout&gt;
      &lt;form method="POST" action="http://localhost:5000/formPart1"&gt;
        &lt;h1 className="f2 pink tc montserrat mb3 mt4"&gt;Your Bio&lt;/h1&gt;
        &lt;StatusBar&gt;
          &lt;Link to="/Form1"&gt;1. Basic Details&lt;/Link&gt;
        &lt;/StatusBar&gt;
        &lt;StatusBar&gt;
          &lt;Link to="/Form2"&gt;2. Further Details&lt;/Link&gt;
        &lt;/StatusBar&gt;
        &lt;StatusBar type="active"&gt;3. Your Bio&lt;/StatusBar&gt;
        &lt;TextBox&gt;
          We will use your bio to find you a reading partner. This is also the
          wording weâ€™ll use to introduce you to your reading partner over email.
        &lt;/TextBox&gt;
        &lt;Label&gt;{q1}&lt;/Label&gt;
        &lt;TextArea placeholder="About You" /&gt;
        &lt;Label&gt;{q2}&lt;/Label&gt;
        &lt;TextArea placeholder="Special Requirements" /&gt;
        &lt;Label&gt;{q3}&lt;/Label&gt;
        &lt;RadioButton&gt;1 (Like the idea)&lt;/RadioButton&gt;
        &lt;RadioButton&gt;2&lt;/RadioButton&gt;
        &lt;RadioButton&gt;3 (Don&amp;apos;t mind) &lt;/RadioButton&gt;
        &lt;RadioButton&gt;4&lt;/RadioButton&gt;
        &lt;RadioButton&gt;5 (Really Don&amp;apos;t Like)&lt;/RadioButton&gt;
        &lt;Link to="/thankyou" className="no-underline"&gt;
          &lt;Button type="register"&gt;Submit&lt;/Button&gt;
        &lt;/Link&gt;
      &lt;/form&gt;
      &lt;TextBox&gt;
        &lt;Link to="/" className="no-underline"&gt;Go back to the homepage&lt;/Link&gt;
      &lt;/TextBox&gt;
    &lt;/Layout&gt;
  )
}
&nbsp;
Form3.propTypes = {
  data: PropTypes.object,
}
&nbsp;
export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 3" } }) {
      frontmatter {
        q1
        q2
        q3
      }
    }
  }
`
&nbsp;
export default Form3
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
