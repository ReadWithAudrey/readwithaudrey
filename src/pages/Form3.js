import React from 'react'
import { Link, graphql } from 'gatsby'
import PropTypes from 'prop-types'

import Layout from '../components/layout'
import { Label, Button, StatusBar, RadioButton, TextBox } from '../components/'

const Form3 = ({ data }) => {
  const { q1, q2, q3 } = data.markdownRemark.frontmatter
  return (
    <Layout>
      <form method="POST" action="http://localhost:5000/formPart1">
        <h1>Further Details</h1>
        <StatusBar>
          <Link to="/Form1">1. Basic Details</Link>
        </StatusBar>
        <StatusBar>
          <Link to="/Form2">2. Further Details</Link>
        </StatusBar>
        <StatusBar type="active">
          <Link to="/Form3">3. Your Bio</Link>
        </StatusBar>
        <p>
          We will use your bio to find you a reading partner. This is also the
          wording weâ€™ll use to introduce you to your reading partner over email.
        </p>
        <Label>{q1}</Label>
        <TextBox placeholder="About You" />
        <Label>{q2}</Label>
        <TextBox placeholder="Special Requirements" />
        <Label>{q3}</Label>
        <RadioButton>1</RadioButton>
        <RadioButton>2</RadioButton>
        <RadioButton>3</RadioButton>
        <RadioButton>4</RadioButton>
        <RadioButton>5</RadioButton>
        <Button>Submit</Button>
      </form>
      <Link to="/">Go back to the homepage</Link>
    </Layout>
  )
}

Form3.propTypes = {
  data: PropTypes.object,
}

export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 3" } }) {
      frontmatter {
        q1
        q2
        q3
      }
    }
  }
`

export default Form3
