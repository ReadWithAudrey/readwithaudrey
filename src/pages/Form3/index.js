import React from 'react'
import { Link, graphql } from 'gatsby'
import PropTypes from 'prop-types'
import { SignupContext } from '../../contexts/SignupContext'

import {
  Label,
  Button,
  StatusBar,
  TextArea,
  TextBox,
  Layout,
  FormSection,
} from '../../components/'

const Form3 = ({ data }) => {
  const { q1, q2 } = data.markdownRemark.frontmatter
  return (
    <Layout>
      <h1 className="f2 pink tc montserrat mb3 mt4">Nearly there…</h1>
      <FormSection>
        <Link to="/form1">
          <StatusBar>1. Basic Details</StatusBar>
        </Link>
        <Link to="/form2">
          <StatusBar>2. Further Details</StatusBar>
        </Link>
        <StatusBar type="active">3. Your Bio</StatusBar>
        <TextBox>
          Please tell us about yourself in up to 100 words. We’ll use your story
          (bio) to match you up with a reading partner. We’ll also use it when
          we introduce you to your reading partner.
        </TextBox>
        <SignupContext.Consumer>
          {({ story, specialRequests, updateForm }) => (
            <form method="POST" action="http://localhost:5000/formPart1">
              <Label>{q1}</Label>
              <TextArea
                placeholder="About You"
                onChange={updateForm}
                name="story"
                value={story}
              />
              <TextBox className="f">
                Tips: The most effective matches tend to happen when members
                share bios that are open and interesting, share a touch of
                quirkiness, and feel light-hearted in spirit. Consider
                describing the things you’re passionate about. It’s helpful to
                share your interests and what you do in your spare time. Try to
                avoid overused phrases e.g. easy-going. Instead, consider what
                makes you special ☺
              </TextBox>
              <Label>{q2}</Label>
              <TextArea
                placeholder="Any special requests for instance"
                onChange={updateForm}
                name="specialRequests"
                value={specialRequests}
              />
              <Link to="/thankyou" className="no-underline">
                <Button type="register">Submit</Button>
              </Link>
            </form>
          )}
        </SignupContext.Consumer>
      </FormSection>
    </Layout>
  )
}

Form3.propTypes = {
  data: PropTypes.object,
}

export const query = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    markdownRemark(frontmatter: { title: { eq: "Form Part 3" } }) {
      frontmatter {
        q1
        q2
      }
    }
  }
`

export default Form3
