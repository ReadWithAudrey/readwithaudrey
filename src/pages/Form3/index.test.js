import React from 'react'
import renderer from 'react-test-renderer'
import Form3 from './index.js'

describe('Form3 page snapshot test', () =>
  it('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'Form Part 3',
        },
      },
      markdownRemark: {
        frontmatter: {
          q1: 'Your Bio',
          q2: 'Special Requirements',
          q3: 'How do you feel about Abridged books?',
        },
      },
    }
    const tree = renderer.create(<Form3 data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
