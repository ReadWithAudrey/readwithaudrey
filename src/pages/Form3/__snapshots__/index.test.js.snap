// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form3 page snapshot test renders correctly 1`] = `
Array [
  <header
    className="sc-bdVaJa flex items-center justify-between ph3 pv2 montserrat white bg-green w-100 f3 gzTCKs"
  >
    <div
      className="flex items-center"
      id="header-left"
    >
      <Link
        to="/"
      >
        <img
          alt="book icon"
          className="h2 ph2"
          src="test-file-stub"
        />
      </Link>
      <Link
        to="/"
      >
        Audrey
      </Link>
    </div>
    <nav
      className="sc-bwzfXH w-65 w-40-l montserrat f4 gQAWUv"
    >
      <section
        className="sc-htpNat montserrat bRYTxq"
      >
        <Link
          to="/story/"
        >
          Why?
        </Link>
        <Link
          to="/faq/"
        >
          FAQ
        </Link>
        <Link
          to="/contact/"
        >
          Contact
        </Link>
        <Link
          className="mv2 white br2 ba bw1 b--white"
          to="/Form1"
        >
          Join Us
        </Link>
      </section>
      <section
        className="sc-bxivhb igYaY"
      >
        <a
          className="dtc fr tr v-mid dn-ns p0-ns link dim white"
          onClick={[Function]}
        >
          <span
            className="db w2 mt1 mb2 bw2 pb2 bt bb b--white"
          />
          <span
            className="db w2 mt1 bw2 pb1 bt b--white"
          />
        </a>
      </section>
    </nav>
  </header>,
  <div
    className="flex flex-column items-center justify-center page-padding ph4"
  >
    <h1
      className="f2 pink tc montserrat mb3 mt4"
    >
      Your Bio
    </h1>
    <section
      className="sc-dxgOiQ b--green cGznzi"
    >
      <Link
        to="/form1"
      >
        <section
          className="sc-gqjmRU flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 bypQGG"
        >
          1. Basic Details
        </section>
      </Link>
      <Link
        to="/form2"
      >
        <section
          className="sc-gqjmRU flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 bypQGG"
        >
          2. Further Details
        </section>
      </Link>
      <section
        className="sc-VigVT fl w-third bg-green montserrat ehcWMy sc-gqjmRU flex items-center fl w-third montserrat bg-moon-gray fw4 f5 mb3 bypQGG"
      >
        3. Your Bio
      </section>
      <p
        className="sc-kAzzGY montserrat f5 tc lh-copy jdrnSq"
      >
        We will use your bio to find you a reading partner. This is also the wording we’ll use to introduce you to your reading partner over email.
      </p>
      <form
        action="http://localhost:5000/formPart1"
        method="POST"
      >
        <label
          className="sc-htoDjs pink montserrat fw9 f5 pv2 kGbKV"
        >
          Your Bio
        </label>
        <textarea
          className="sc-kpOJdX montserrat eZKhxe"
          placeholder="About You"
          rows="5"
        />
        <label
          className="sc-htoDjs pink montserrat fw9 f5 pv2 kGbKV"
        >
          Special Requirements
        </label>
        <textarea
          className="sc-kpOJdX montserrat eZKhxe"
          placeholder="Special Requirements"
          rows="5"
        />
        <label
          className="sc-htoDjs pink montserrat fw9 f5 pv2 kGbKV"
        >
          How do you feel about Abridged books?
        </label>
        <section
          className="sc-fjdhpX montserrat fw4 f5 pv2 eQHpIT"
        >
          <input
            className="sc-jzJRlG fsyDfV"
            name="1 (Like the idea)"
            type="radio"
            value="1 (Like the idea)"
          />
          <span
            className="sc-cSHVUG bVIEwX"
          >
            1 (Like the idea)
          </span>
        </section>
        <section
          className="sc-fjdhpX montserrat fw4 f5 pv2 eQHpIT"
        >
          <input
            className="sc-jzJRlG fsyDfV"
            name="2"
            type="radio"
            value="2"
          />
          <span
            className="sc-cSHVUG bVIEwX"
          >
            2
          </span>
        </section>
        <section
          className="sc-fjdhpX montserrat fw4 f5 pv2 eQHpIT"
        >
          <input
            className="sc-jzJRlG fsyDfV"
            name="3 (Don't mind) "
            type="radio"
            value="3 (Don't mind) "
          />
          <span
            className="sc-cSHVUG bVIEwX"
          >
            3 (Don't mind) 
          </span>
        </section>
        <section
          className="sc-fjdhpX montserrat fw4 f5 pv2 eQHpIT"
        >
          <input
            className="sc-jzJRlG fsyDfV"
            name="4"
            type="radio"
            value="4"
          />
          <span
            className="sc-cSHVUG bVIEwX"
          >
            4
          </span>
        </section>
        <section
          className="sc-fjdhpX montserrat fw4 f5 pv2 eQHpIT"
        >
          <input
            className="sc-jzJRlG fsyDfV"
            name="5 (Really Don't Like)"
            type="radio"
            value="5 (Really Don't Like)"
          />
          <span
            className="sc-cSHVUG bVIEwX"
          >
            5 (Really Don't Like)
          </span>
        </section>
        <Link
          className="no-underline"
          to="/thankyou"
        >
          <button
            className="sc-gZMcBi bg-pink hover-bg-orange white pa2 br2 w-50 h3 f3 mt4 center gTtpNx"
          >
            Submit
          </button>
        </Link>
      </form>
    </section>
  </div>,
  <footer
    className="sc-EHOje flex flex-column justify-center items-center montserrat bg-green w-100 bt bw2 b--green jvzENE"
  >
    <span
      className="sc-bZQynM flex justify-around items-center bg-white black w-100  pa2 jrqyAQ"
    >
      © Audrey
      <Link
        className="no-underline black ph1"
        to="/terms/"
      >
        Terms
      </Link>
      <Link
        className="no-underline black ph1"
        to="/privacy/"
      >
        Privacy
      </Link>
      <Link
        className="no-underline black ph1"
        to="/contact/"
      >
        Contact
      </Link>
    </span>
  </footer>,
]
`;
